src/Jackett.Common/Indexers/GimmePeers.cs

```diff
@@ -134,7 +134,7 @@ protected override async Task<IEnumerable<ReleaseInfo>> PerformQuery(TorznabQuer
                 var parser = new HtmlParser();
                 var dom = parser.ParseDocument(results);
 
-                var rows = dom.QuerySelectorAll(".browsetable")?.Last().QuerySelectorAll("tr");
+                var rows = dom.QuerySelectorAll(".browsetable").LastOrDefault()?.QuerySelectorAll("tr");
```

Comment:
Are you sure default is null? We had problems with this kind of things in the past.
Also, the next line is not going to work because rows will be a list, not null. Don't try to do all things in one line...

Reply:
`default(T) is null where T : object`. Only value types (int, bool, long, DateTime, struct) are non-null. https://www.c-sharpcorner.com/article/defaultt-in-generics/

The problem we had was we used `FirstOrDefault` when we only wanted to use First when it was the only element in the list. It wasn't that the default wasn't null in that case, it was that we weren't collecting the correct value when `Count > 1` which FirstOrDefault wouldn't return default for Count > 0.

The null conditional operator `?.` short circuits any evaluation to return `null` if the expression behind it is `null`. So the second line will work if `LastOrDefault()` is used on an empty list. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/operators/member-access-operators#null-conditional-operators--and-

I wanted to keep this a single line for now, because it's a blind fix, we have no devs or regular testers to make sure any other refactorings are working. We're getting an exception from Last because there is nothing to return for last in the case of no results, so we return Default in that case which is `null` for all nullable types (putting the `?.` operator on a non-nullable type would lead to compile errors), check for the null with null conditional so we don't get a null reference exception on the next `QuerySelectorAll` call, and check for null in the next line.

When we get an invite so we can look at the source, I want to find a completely different CSS selector to use because this whole thing seems like a hack, and probably is since this is one of our blind conversions from CsQuery to AngleSharp.

Reply:
All clear.